src/parser/choice.rs

```diff
@@ -775,27 +775,27 @@ macro_rules! dispatch_parser_impl {
 
 #[macro_export]
 macro_rules! dispatch_inner {
-    ($expr_ident: ident [$first_ident: ident $($id: ident)*] [$($collected: tt)*] $($pat: pat)|+ => $expr: expr, $($rest_alt: tt)*) => {
-        $crate::dispatch_inner!{ $expr_ident [ $($id)* ] [$($collected)* $first_ident $($pat)|+ => $expr,] $($rest_alt)*}
+    ($expr_ident: ident [$first_ident: ident $($id: ident)*] [$($collected: tt)*] $($pat: pat)|+ $(if $pred:expr)* => $expr: expr, $($rest_alt: tt)*) => {
```

Comment:
```suggestion
    ($expr_ident: ident [$first_ident: ident $($id: ident)*] [$($collected: tt)*] $($pat: pat)|+ $(if $pred:expr)? => $expr: expr, $($rest_alt: tt)*) => {
```

The guard ought to only appear once or never so `?` would be more appropriate